### ISSUE TITLE
Should warn when typing.NamedTuple uses mutable defaults

### ISSUE THREAD

**Issue by Seanny123** [PyCQA/pylint#3716]
## Problem

Giving mutable default arguments to `typing.NamedTuple` can cause unexpected behaviour.
```python
from typing import List, NamedTuple

class LibStats(NamedTuple):
    avgs: List[float] = []
    stds: List[float] = []

lib_a = LibStats()
lib_b = LibStats()

lib_a.avgs.append([1.])
# this assert fails, because list is shared between the instances
assert lib_a.avgs != lib_b.avgs
```

The same occurs when [providing a dictionary as a default argument](https://stackoverflow.com/q/59205528/1079075).

## Expected Behaviour

`pylint` should warn about this code, similar to how it currently warn about mutable objects being used as default arguments in function definitions.

### Additional context
pylint version: 2.5.3

**Comment by AWhetter** [comment]
Agreed. We should make `dangerous-default-value` check for this case.

**Comment by skasch** [comment]
I ran into the same issue with older versions of Python / `namedtuple`. Here's the snippet I used to test that:

```py
# -*- coding: utf-8 -*-
"""The namedtuple_default script."""

from collections import namedtuple
from typing import List, NamedTuple, Optional, Text

MyNamedTuple = namedtuple("MyNamedTuple", ["foo", "bar"])
MyNamedTuple.__new__.__defaults__ = (None, [])

MyTypedNamedTuple = NamedTuple("MyTypedNamedTuple", [("foo", Optional[Text]), ("bar", List[int])])
MyTypedNamedTuple.__new__.__defaults__ = (None, [])
```

This is the [recommended way in Stackoverflow](https://stackoverflow.com/questions/11351032/named-tuple-and-default-values-for-optional-keyword-arguments) to set default values to a `namedtuple` for Python versions <3.7. Tested with Python 2.7, 3.7, and 3.9.

**Comment by vapier** [comment]
use dataclass w/default_factory to handle this safely.
https://docs.python.org/3/library/dataclasses.html#dataclasses.field

requires Python 3.8+, or the dataclass module.

**Comment by nathanielmanistaatgoogle** [comment]
Another sample program for this issue (similar to what it seen in [this earlier comment](https://github.com/pylint-dev/pylint/issues/3716#issuecomment-811925921), but shorter and avoiding the dunders):
```python
import collections

class BadClass(
    collections.namedtuple(
        'BadClass', ('values',), defaults=([],))):
    """Has a problematically-mutable default value for its field."""
```
